src/experimental/Db.h

```diff
@@ -51,112 +51,58 @@ class Entry {
 // constructor
 class PilotEntry : public Entry {
     PilotEntry() = delete;
+public:
     PilotEntry(int row_id) : Entry::Entry(DataPosition("pilots", row_id)) {}
-};
-
-Entry newPilotEntry(int row_id) { return Entry(DataPosition("pilots", row_id)); }
-
-static
-bool insertPilot(UserInput& uin)
-{
-    DEB("Inserting...");
-    auto data = uin.all();
-    auto position = DEFAULT_PILOT_POSITION;
-
-    if (data.isEmpty()) {
-        DEB("Object Contains no data. Aborting.");
-        return false;
-    }
-    QString statement = "INSERT INTO " + position.first + QLatin1String(" (");
-    for (auto i = data.cbegin(); i != data.cend(); ++i) {
-        statement += i.key() + QLatin1String(", ");
+    PilotEntry(TableData fromNewPilotDialog) : Entry::Entry(DataPosition("pilots", 0)) {
+        setData(fromNewPilotDialog);
```

Comment:
This here is exactly what i mention about the position. If we are getting data from the dialog we *should* also have the position. It makes sense to do so since we populate the gui with entries from the database. We *know* where they are!

Reply:
didn't see that comment, disregard my question in the comment above